lib/modules.nix

```
@@ -115,8 +115,21 @@ rec {
 
       checkUnmatched =
         if config._module.check && config._module.freeformType == null && merged.unmatchedDefns != [] then
-          let inherit (head merged.unmatchedDefns) file prefix;
-          in throw "The option `${showOption prefix}' defined in `${file}' does not exist."
+          let
+            inherit (head merged.unmatchedDefns) file prefix;
+            filteredSet = builtins.removeAttrs (attrByPath (sublist 0 (length prefix - 1) prefix) {} options)
+              [ "_module" ];
+            baseMsg = "The option `${showOption (prefix)}' defined in `${file}' does not exist.";
+          in
+            if filteredSet == {}
+              then throw ''
+                ${baseMsg}
+
+                However it seems as there are no options defined in ${showOption prefix}. Are you sure you've
+                declared your options properly? This happens if you e.g. declared your options in `types.submodule'
+                under `config' rather than `options'.
```

From looking at this, I just noticed that in #82743 I introduced a mistake in the error message for missing options! I created https://github.com/NixOS/nixpkgs/pull/95718 to fix it again, which will unfortunately give a merge conflict again. Alternatively you could incorporate the commit from that PR into this one.

With that changed, I suggest to change the error message to
```suggestion
                However there are no options defined in ${showOption prefix}. Are you sure you've
                declared your options properly? This can happen if you declared your options in `types.submodule'
                under `config' rather than `options'.
```

(note that this should be `prefix` and not `firstDef.prefix` from my PR)

---

Ah nice :-)
Rebased and updated accordingly.